### PR TITLE
CMake update

### PR DESCRIPTION
**Pull request by Hs293Go**
- Make including QORE directories conditional: Addresses  #1062
- Replace find_package with modern find_dependency in acadosConfig.cmake to let acados propagate find_package args like REQUIRED
- Installs acados's own Find*.cmake scripts to let them be found if the user installs to non-default locations by tweaking ACADOS_INSTALL_DIR
- Remove unused `FindNumpy.cmake`, `FindMatlab.cmake`

### CODE REVIEW

**Comment by Hs293Go** [comment]
A second part of this PR addresses a similar issue of optional dependencies leaking and showing warnings to users that consume acados, i.e. `find_package(FortranLibs)` and `find_package(OpenBLAS)` are unconditional inside `acadosConfig.cmake.in` 

https://github.com/acados/acados/blob/66bd329cd7b512bda697abb8cb43401bfb15361a/cmake/acadosConfig.cmake.in#L73C1-L73C23

even though they appear to be needed by `ooqp` only.

Furthermore, `find_package` calls inside `acadosConfig.cmake.in` is modernized to `find_dependency` calls such that find_package args like REQUIRED and QUIET will be propagated. 

Finally, logic is added in `acados/CMakeLists.txt` to install `FindFortranLibs.cmake` and `FindOpenBLAS.cmake` (and other Find... scripts). Otherwise these scripts will not be found if the user tweaks `ACADOS_INSTALL_DIR` and installs acados to other locations, e.g. the system `/usr/local`

**Review by FreyJo — COMMENTED** [review]
Thanks for the PR!
I left two comments.

File: acados/CMakeLists.txt [hunk 1]
```diff
@@ -72,12 +72,12 @@ add_library(acados ${ACADOS_SRC})
 target_include_directories(acados
     PUBLIC
         $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}> # to be able to include "acados/header.h"
-        $<BUILD_INTERFACE:${EXTERNAL_SRC_DIR}/qore/QPSOLVER_DENSE/include>
-        $<BUILD_INTERFACE:${EXTERNAL_SRC_DIR}/qore/QPSOLVER_DENSE/source>
-        $<BUILD_INTERFACE:${EXTERNAL_SRC_DIR}/qore/KKTPACK_DENSE/include>
-        $<BUILD_INTERFACE:${EXTERNAL_SRC_DIR}/qore/KKTPACK_DENSE/source>
-        $<BUILD_INTERFACE:${EXTERNAL_SRC_DIR}/qore/QPCORE/include>
-        $<BUILD_INTERFACE:${EXTERNAL_SRC_DIR}/qore>
+        $<$<BOOL:${ACADOS_WITH_QORE}>:$<BUILD_INTERFACE:${EXTERNAL_SRC_DIR}/qore/QPSOLVER_DENSE/include>>
```
[the remark this answers is not in the document]
Comment (Hs293Go) [reply acados/CMakeLists.txt]: I agree. I originally went with the nested generator to minimize diffs, but it's more than fair to prioritize feasibility.

**Comment by Hs293Go** [comment]
I complied with the changes you suggested. I had been conservative in wanting to minimize diffs.

Changes were divided in two commits was because initially I wanted to just fix qore include dirs only. But then I skimmed the forum and issues and found #1047 and some other issues about people wanting to tweak their Acados install location e.g. system-wide install to /usr/local, which would require cooperation from find_package. I judged the time is ripe and took the liberty to squeeze in a second commit in this PR.

**Review by FreyJo — APPROVED** [review]
Updated the PR description, looks all good now :+1:

**Pull request merged**